test/test_pause.js

```diff
@@ -20,338 +20,379 @@ describe('.pause', () => {
-    this.timeout(10000);
-    const queue = new Queue('test');
+    it('pauses fast when queue is drained', function(done) {
+      this.timeout(10000);
+      const queue = new Queue('test');
 
-    queue.process((/*job*/) => {
-      Promise.resolve();
-    });
+      queue.process((/*job*/) => {
+        Promise.resolve();
+      });
 
-    queue.add({});
+      queue.add({});
 
-    queue.on('drained', () => {
-      delay(500).then(() => {
-        const start = new Date().getTime();
-        return queue.pause(true).finally(() => {
-          const finish = new Date().getTime();
-          expect(finish - start).to.be.lt(1000);
-          queue.close().then(done, done);
+      queue.on('drained', () => {
+        delay(500).then(() => {
+          const start = new Date().getTime();
+          return queue.pause(true).finally(() => {
+            const finish = new Date().getTime();
+            expect(finish - start).to.be.lt(1000);
+            queue.close().then(done, done);
+          });
         });
       });
     });
-  });
 
-  it('should not processed delayed jobs', function(done) {
-    this.timeout(5000);
-    const queue = new Queue('pause-test');
+    describe('with doNotWaitActive=true', () => {
+      it('should not wait for active jobs to finish', async () => {
+        const queue = utils.buildQueue();
+        await queue.add({});
 
-    queue.process(() => {
-      done(new Error('should not process delayed jobs in paused queue.'));
-    });
+        // wait for us to start processing job
+        await new Promise(resolve => {
+          queue.process(() => {
+            // resolve promise, but continue processing job forever
+            resolve();
 
-    queue.pause().then(() => {
-      queue
-        .add(
-          {},
-          {
-            delay: 500
-          }
-        )
-        .then(() => {
-          return queue.getJobCounts();
-        })
-        .then(counts => {
-          expect(counts).to.have.property('paused', 0);
-          expect(counts).to.have.property('waiting', 0);
-          expect(counts).to.have.property('delayed', 1);
-          return delay(1000);
-        })
-        .then(() => {
-          return queue.getJobCounts();
-        })
-        .then(counts => {
-          expect(counts).to.have.property('paused', 1);
-          expect(counts).to.have.property('waiting', 0);
-          done();
+            return new Promise(() => {});
+          });
         });
+
+        return queue.pause(true, true);
+      });
+
+      it('should not process new jobs', async () => {
+        const queue = utils.buildQueue();
+
+        // wait for processor to be online
+        const jobProcessor = sinon.stub();
+        queue.process(jobProcessor);
+
+        await delay(1000);
+        await queue.pause(true, true);
+
+        // Add another job. This should not be processed
+        await queue.add({});
+
+        await delay(1000);
```

Comment:
Let me know if there's a better way to do this other than wait an arbitrary amount of time to see if a job got processed.